esp-hal-procmacros/src/lp_core.rs

```diff
@@ -23,19 +23,19 @@ pub fn entry(args: TokenStream, input: TokenStream) -> TokenStream {
         let mut res = String::from("__ULP_MAGIC_");
         for &a in args {
             let t = &a.ty;
-            let quoted = to_string(&t);
+            let quoted = to_string(t);
             res.push_str(&quoted);
-            res.push_str("$");
+            res.push('$');
         }
 
         res
     }
 
     pub(crate) fn get_simplename(t: &Type) -> String {
-        String::from(match t {
+        match t {
             Type::Path(p) => String::from(&p.path.segments.last().unwrap().ident.to_string()),
```

Comment:
I'll confess that this is a guess 😅 but based on what I've seen in other files it should work
```suggestion
            Type::Path(p) => p.path.segments.last().unwrap().ident.to_string(),
```